aderyn_core/src/detect/experimental/demo_parent_chain.rs

```diff
@@ -0,0 +1,126 @@
+use std::{collections::BTreeMap, error::Error};
+
+use crate::{
+    ast::{self, NodeID},
+    capture,
+    context::{
+        browser::GetParentChain,
+        workspace_context::{ASTNode, WorkspaceContext},
+    },
+    detect::detector::{IssueDetector, IssueDetectorNamePool, IssueSeverity},
+};
+use eyre::Result;
+
+#[derive(Default)]
+pub struct ParentChainDemonstrator {
+    // Keys are source file name and line number
+    found_instances: BTreeMap<(String, usize), NodeID>,
+}
+
+/*
+
+In ParentChainContract.sol, there is only 1 assignment done. The goal is to capture it first.
+Then walk up it's parent tree and capture a) contract definition b) if statement c) for statement
```

Comment:
This file shows an example of how you could traverse the AST upward from a given node